nucliadb/nucliadb/common/maindb/tikv.py

```diff
@@ -80,9 +84,17 @@ async def batch_get(self, keys: List[str]):
         with tikv_observer({"type": "batch_get"}):
             return await self.txn.batch_get(bytes_keys)
 
+    @backoff.on_exception(backoff.expo, (TimeoutError,), max_tries=2)
     async def get(self, key: str) -> Optional[bytes]:
         with tikv_observer({"type": "get"}):
-            return await self.txn.get(key.encode())
+            try:
+                return await self.txn.get(key.encode())
+            except Exception as exc:
+                exc_text = str(exc)
+                if "4-DEADLINE_EXCEEDED" in exc_text:
```

Comment:
nit: if possible let's not rely on the message text and use the constant (I assume you have that somewehere in the exc object)

```
>>> import grpc
>>> grpc.StatusCode.DEADLINE_EXCEEDED
<StatusCode.DEADLINE_EXCEEDED: (4, 'deadline exceeded')>
```

Reply:
The client is funky. It's rust grpc under the hood but only throws raw base `Exception` errors with only strings stuffed inside

Reply:
let me put up a comment! sorry, should have mentioned hehe